tree/treeplayer/inc/ROOT/TDFActionHelpers.hxx

```diff
@@ -483,15 +488,16 @@ class SnapshotHelperMT {
    std::vector<int> fIsFirstEvent;    // vector<bool> is evil
    const std::string fDirName;        // name of TFile subdirectory in which output must be written (possibly empty)
    const std::string fTreeName;       // name of output tree
+   const SnapshotOptions fOptions;    // struct holding options to pass down to TFile and TTree in this action
    const ColumnNames_t fBranchNames;
 
 public:
    using BranchTypes_t = TypeList<BranchTypes...>;
    SnapshotHelperMT(const unsigned int nSlots, std::string_view filename, std::string_view dirname,
-                    std::string_view treename, const ColumnNames_t &bnames)
-      : fNSlots(nSlots), fMerger(new ROOT::Experimental::TBufferMerger(std::string(filename).c_str(), "RECREATE")),
+                    std::string_view treename, const ColumnNames_t &bnames, SnapshotOptions const &options)
```

Comment:
here and below it should be `const SnapshotOptions &` for consistency (in the case of references `const T&` and `T const &` are synonyms so there can be no confusion about what you mean anyway)

Reply:
I think the `T const &x` form is what clang-format likes and/or is part of the coding conventions of ROOT, but I can cange to `const T &x`. I also like that better and it makes sense to keep things consistent.

Reply:
clang-format is fine with `const T &x`, as long as the `&` is attached to the variable name and not to the type

Reply:
Sure, I will change it. I actually like more the way you suggested.